### DISCUSSION TITLE
Why if using `MetaMaskConnector`, when disconnecting the wallet event `watchNetwork` triggers ?

### DISCUSSION THREAD

**Discussion by ghost** [wevm/wagmi#1680] [Q&A]
When I disconnect my MetaMask wallet, event `watchNetwork` triggers. Although logically it should trigger `watchAccount` event.

**Comment by tmm** [comment]
My guess is it is related to this long-running MetaMask bug: https://github.com/MetaMask/metamask-extension/issues/13375
